web/controllers/course_controller.ex

```diff
@@ -0,0 +1,65 @@
+defmodule CoursePlanner.CourseController do
+  use CoursePlanner.Web, :controller
+
+  alias CoursePlanner.Course
+
+  def index(conn, _params) do
+    courses = Repo.all(Course)
+    render(conn, "index.html", courses: courses)
+  end
+
+  def new(conn, _params) do
+    changeset = Course.changeset(%Course{})
+    render(conn, "new.html", changeset: changeset)
+  end
+
+  def create(conn, %{"course" => course_params}) do
+    changeset = Course.changeset(%Course{}, course_params, :create)
+
+    case Repo.insert(changeset) do
+      {:ok, _course} ->
+        conn
+        |> put_flash(:info, "Course created successfully.")
+        |> redirect(to: course_path(conn, :index))
+      {:error, changeset} ->
+        render(conn, "new.html", changeset: changeset)
+    end
+  end
+
+  def show(conn, %{"id" => id}) do
+    course = Repo.get!(Course, id)
+    render(conn, "show.html", course: course)
+  end
+
+  def edit(conn, %{"id" => id}) do
+    course = Repo.get!(Course, id)
+    changeset = Course.changeset(course)
+    render(conn, "edit.html", course: course, changeset: changeset)
+  end
+
+  def update(conn, %{"id" => id, "course" => course_params}) do
+    course = Repo.get!(Course, id)
+    changeset = Course.changeset(course, course_params)
+
+    case Repo.update(changeset) do
+      {:ok, course} ->
+        conn
+        |> put_flash(:info, "Course updated successfully.")
+        |> redirect(to: course_path(conn, :show, course))
+      {:error, changeset} ->
+        render(conn, "edit.html", course: course, changeset: changeset)
+    end
+  end
+
+  def delete(conn, %{"id" => id}) do
+    course = Repo.get!(Course, id)
+
+    # Here we use delete! (with a bang) because we expect
+    # it to always work (and if it does not, it will raise).
+    Repo.delete!(course)
```

Comment:
for this, shall I just set the status to deleted ?
and then in the list to show only those that are not having "deleted" as status ?